include/llama/Array.hpp

```diff
@@ -142,15 +142,15 @@ struct Array
     push_front( T const new_element ) const
     -> Array<
         T,
-        count+1
+        count + 1
     >
     {
         Array<
             T,
-            count+1
+            count + 1
         > result;
         for( std::size_t i = 0; i < count - 1; i++ )
-            result.element[ i + 1 ] = element[ i ];
+            result.element[ count - i ] = element[ count - 1 - i ];
```

Comment:
Why do you copy from back to front? IMO only the loop was wrong and must be `for( std::size_t i = 0; i < count; i++ )` since the array has `count + 1` elements.

Reply:
I would say even in you variant you miss one element.

Reply:
Because I considered it an in-place copy. But its not, thus we can keep the forward copy. Will fix.